### ISSUE TITLE
Do something about long initial load

### ISSUE THREAD

**Issue by vgmoose** [fortheusers/hb-appstore#25]
The initial load takes a while because it has to fetch all icon/banner images for every app, and as more apps are added to the store this load becomes slower and slower.

Some possible solutions:
- try to download images in parallel instead of sequentially
- download a zip of all the images if a significant portion of the images are missing (would need #24 and #5 fixed though)
- download images "on demand" when they are viewed in the list/detailes view instead of during the first load (and in another thread)
- "just" have networking in general faster be faster on the switch (pretty sure homebrew network speed isn't as fast as nintendo's speeds, but have no proof to back up this claim)

As of 2.0, it does re-use the http socket, which has helped a bit, but it's still too slow for the first load

**Comment by vgmoose** [comment]
I've tried downloading a zip of images, and it seems to be a wash in terms of extraction time vs download time.

I think the real mistake of the initial load was having banners be as large as they are, and loaded at the start. On Wii U, we've skipped the banners altogether and the initial load is much less frustrating.

As a rest of this, I think the only true solution to this issue is the third bullet note, downloading them "on demand" in a few worker threads, and upon visiting the app details page in the background.

This should also open up more flexibility like having multiple screenshots in the details view per app.
